dev-core

```diff
@@ -1,16 +1,49 @@
 #!/bin/bash
 
-echo "Building mesg/core:local..."
-docker build -t mesg/core:local --build-arg version="local" .
+set -e
+
+DIR=${GOPATH:-$HOME/go}
+VERSION=local
+LDFLAGS="-X 'github.com/mesg-foundation/core/version.Version=$VERSION'"
+
+echo "compile cli and core"
+GOOS=linux GOARCH=amd64 go build -o ./bin/core -ldflags="$LDFLAGS" core/main.go
+go build -o ./bin/cli -ldflags="$LDFLAGS" interface/cli/main.go
+
+CORE_SUM_PATH="./bin/.core.sum"
+touch "$CORE_SUM_PATH"
+
+# check if core bin was is cached
+CORE_SUM="$(openssl md5 ./bin/core)"
+CORE_SUM_PREV="$(cat $CORE_SUM_PATH)"
+if [[ "$CORE_SUM" == "$CORE_SUM_PREV" ]]; then
+  BINCACHED=true
+else
+  echo "$CORE_SUM" > "$CORE_SUM_PATH"
+fi
 
 # create a symlink for system services.
-rm -rf ~/.mesg/systemservices
-mkdir -p ~/.mesg
-ln -s $GOPATH/src/github.com/mesg-foundation/core/systemservices/sources ~/.mesg/systemservices
+rm -rf "$HOME/.mesg/systemservices"
+mkdir -p "$HOME/.mesg/"
+ln -s "$DIR/src/github.com/mesg-foundation/core/systemservices/sources" "$HOME/.mesg/systemservices"
+
+echo "build mesg/core image"
+if [[ ! $BINCACHED ]]; then
+  docker build -f Dockerfile.dev -t "mesg/core:$VERSION" .
+fi
+
+# remove docker core service if it wasn't clean up.
+COREID="$(docker service ls --format="{{.ID}} {{.Name}}" | grep core | awk '{{print $1}}')"
```

Comment:
It's better if we can also _docker rm_ the containers of removed services. Because sometimes Core things that if it can find a service's container then service is still running and it won't start the service when requested.

Reply:
> Because sometimes Core things that if it can find a service's container then service is still running and it won't start the service when requested.

What is core things? I don't get what you mean.

Reply:
Yes it should be. But in the case where Core removes services we had this issue https://github.com/mesg-foundation/core/issues/536. Still, sometimes I required to execute this `docker rm -f $(docker ps -a -q)` command to get rid it. We should second check if the issue still persist.

Reply:
ok, I do I know which container was run by core services? Is it possible to know it, do we have some labels? If we have then I will add it right away, if not then I don't `docker rm -f $(docker ps -a -q)` because it will remove everything.

Reply:
Yes we have labels for containers as well.

Reply:
Done